actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -59,7 +59,7 @@ def check_path_parameters!
       path_parameters.each do |key, value|
         next unless value.respond_to?(:valid_encoding?)
         unless value.valid_encoding?
-          raise ActionController::BadRequest, "Invalid parameter: #{key} => #{value}"
```

Comment:
Not exactly. The way [`BadRequest`](https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_controller/metal/exceptions.rb#L5-L15) is designed, it acts as a wrapper for other exceptions. For example see it's intented usage [here](https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_dispatch/http/request.rb#L302) and [here](https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_dispatch/http/request.rb#L310).

By passing only the message as the `type` argument in this line, the actual message is never printed since the exception returns early [here](https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_controller/metal/exceptions.rb#L9).

That said, I understand that this change breaks backwards compatibility since people may already be rescuing  `BadRequest` for encoding errors. So an alternative, backwards-compatible solution could be to update `BadRequest` in order to handle such cases.

What do you think?